File: C/面试题17.21.c
#include<stdio.h>
int trap(int* height, int heightSize){
    int sum=0, volum=0,high=1,left=0,right=heightSize-1;
    for(int i=0;i<heightSize;i++){
        sum += height[i];
    }
    while(left <= right){
        while(left<=right&&height[left]<high)left++;
        while(left<=right&&height[right]<high)right++;
        volum+=right-left+1;
        high++;
    }
    return sum - volum;
}

File: Java/面试题17.21.java
class Solution {
    public int trap(int[] height) {
        int high =1, sum=0,volum=0,left=0,right=height.length-1;
        for(int i=0;i<height.length;i++){
            sum+=height[i];
        }
        while(left<=right){
            while(left<=right&&height[left]<high)left++;
            while(left<=right&&height[right]<high)right--;
            high++;
            volum+=right-left+1;
        }
        return volum - sum;
    }
}